addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -755,6 +755,252 @@ export function preserveCursor(document) {
+//------------------------------------------------------------------------------
+
+const formatsInfos = {
+    italic: {
+        tagName: 'em',
+        isFormatted: isItalic,
+        isTag: (node) => ['EM', 'I'].includes(node.tagName),
+        hasStyle: (node) => Boolean(node.style && node.style['font-style']),
+        addStyle: (node) => node.style['font-style'] = 'italic',
+        addNeutralStyle: (node) => node.style['font-style'] = 'normal',
+        removeStyle: (node) => removeStyle(node, 'font-style'),
+    },
+    bold: {
+        tagName: 'strong',
+        isFormatted: isBold,
+        isTag: (node) => ['STRONG', 'B'].includes(node.tagName),
+        hasStyle: (node) => Boolean(node.style && node.style['font-weight']),
+        addStyle: (node) => node.style['font-weight'] = 'bolder',
+        addNeutralStyle: (node) => {
+            node.style['font-weight'] = 'normal'
+        },
+        removeStyle: (node) => removeStyle(node, 'font-weight'),
+    },
+    underline: {
+        tagName: 'u',
+        isFormatted: isUnderline,
+        isTag: (node) => node.tagName === 'U',
+        hasStyle: (node) => node.style && node.style['text-decoration-line'].includes('underline'),
+        addStyle: (node) => addStyleItem(node, 'text-decoration-line', 'underline'),
+        removeStyle: (node) => removeStyleItem(node, 'text-decoration-line', 'underline'),
+    },
+    strikeThrough: {
+        tagName: 's',
+        isFormatted: isStrikeThrough,
+        isTag: (node) => node.tagName === 'S',
+        hasStyle: (node) => node.style && node.style['text-decoration-line'].includes('line-through'),
+        addStyle: (node) => addStyleItem(node, 'text-decoration-line', 'line-through'),
+        removeStyle: (node) => removeStyleItem(node, 'text-decoration-line', 'line-through'),
+    },
+    fontSize: {
+        isFormatted: isFontSize,
+        hasStyle: (node, props) => node.style && node.style['font-size'] === props.size,
+        addStyle: (node, props) => node.style['font-size'] = props.size,
+        removeStyle: (node) => removeStyle(node, 'font-size'),
+    },
+    switchDirection: {
+        isFormatted: isDirectionSwitched,
+    }
+
+}
+
+const isBlockOrHasClass = (node) => {
+    return isBlock(node) || (node.classList && Boolean(node.classList.length));
+}
+const ancestorsUntilBlockOrHasClass = (node) => {
+    if (!node || !node.parentElement || isBlockOrHasClass(node)) return [];
+    return [node.parentElement, ...ancestorsUntilBlockOrHasClass(node.parentElement)];
+};
+const removeStyle = (node, styleName) => {
+    node.style[styleName] = null;
+    if (node.getAttribute('style') === '') {
+        node.removeAttribute('style');
+    }
+};
+const removeStyleItem = (node, styleName, item) => {
+    const newStyle = node.style[styleName].split(' ').filter(x => x !== item).join(' ');
+    node.style[styleName] = newStyle || null;
+    if (node.getAttribute('style') === '') {
+        node.removeAttribute('style');
+    }
```

Comment:
I suggest to merge this function into `removeStyle`. If an item is provided, it only removes the provided item. Otherwise, it removes the entire style.